verl/utils/config.py

```diff
@@ -14,10 +14,21 @@
 
 from typing import Dict
 
-from omegaconf import DictConfig
+from omegaconf import DictConfig, open_dict
 
 
 def update_dict_with_config(dictionary: Dict, config: DictConfig):
     for key in dictionary:
         if hasattr(config, key):
             dictionary[key] = getattr(config, key)
+
+
+def config_normalize_batch_size(config, key: str, divider: int):
```

Comment:
I'm interested to know why in-place normalization might lead to confusion and how you feel when reading and using veRL's batch size configuration.

Additionally, I'd like to hear about any bugs you may have encountered while using in-place modification.

Reply:
As a simple example:

```
def __init__():
  f()
  ...
  normalize_batch_size()
  ...
  g()

def f():
  ...
  f2()

def f2():
  ...
  f3()

def f3():
  utilize config.batch_size

def g():
  utilize config.batch_size
```

Then, inside f3 we are using the un-normalized batch size, while in g we use the normalized one.
 
Surely this is just a made-up artificial case, but in the real world I am worried whether, in the future, we will read batch size somewhere deeply nested, and accidentally it is *before* the normalize_batch_size, and then we are in troduble.

Reply:
Got it. I think your refactor regarding the in-place normalization is great.

However, I wonder how we deal with more complicated config mutate scenarios (such as $\times$ n in PR https://github.com/volcengine/verl/pull/80) if we follow the no-in-place mutate principle.

It would be better to keep only one additional config for each \*_batch_size config and we may need to make the normalize config function more general

Reply:
> we may need to make the normalize config function more general

Totally agree! What about `config_normalize_batch_size(config, 'batch_size', self.config.rollout.n / self.device_mesh.shape[0])`, and at the same time change "divider" to "multiplier"?
(We may not face rounding issues since divider is always power of 2)